File: tools/improve/tests/original/maudec/py/palindrome.py
import enum

class List(enum.Enum):
	A = 0  # 
	E = 1  # 
	I = 2  # 
	O = 3  # 
	U = 4  # 


def palindrome(S: [List]) -> bool:
	match S:
		case []:
			return True

		case [_]:
			return True

		case [S, *L, S_2] if S == S_2:
			return palindrome(L)

		case [_, *_, _]:
			return False

File: tools/improve/tests/original/maudec/rust/palindrome.rs


#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum List {
	A,
	E,
	I,
	O,
	U,
}

fn palindrome(s: &[List]) -> bool {
	match s {
		[] => true,
		[_] => true,
		[s, l @ .., s_2] if (*s).clone() == (*s_2).clone() => palindrome(l),
		[_, .., _] => false,
		_ => panic!("case not covered"),
	}
}

